### ISSUE TITLE
Cancelled connections

### ISSUE THREAD

**Issue by ShiroYacha** [OpendataCH/Transport#183]
Hi,

This is more like a question then an issue. Is there anyway to identify cancelled connections? I have read that you are merging your info with GTFS-RT from transport.opendata.swiss, but seems they do not provide any information on service alerts...

Thanks!

**Comment by helbling** [comment]
We from timetable.search.ch are still developing the gtfsrt integration.
This turned out to be quite complicated.

We expect it to be ready soon. Cancelled connections are supported.
They even have dynamic additional trips which are often necessary when there is some replacement service.

What do you mean by service alerts?

**Comment by ShiroYacha** [comment]
Hi, I mean in this document https://opentransportdata.swiss/de/cookbook/gtfs-rt/ they only provide trip update but no service alert which is basically cancelled trains etc... Which gtfsrt data/api are you guys using for integration?

**Comment by helbling** [comment]
Yes, we are using this gtfsrt api. Cancelled trains are specified in "trip updates" as well (don't know if they have them for all the transport companies they support in gtfsrt). 

We also have something similar to the service alerts which is this system: http://fahrplan.sbb.ch/bin/help.exe/en?tpl=showmap_external 
Look for the key disruptions in the json returned by the timetable.search.ch api.
This key is only returned whenever there is some disruption at all.

**Comment by ShiroYacha** [comment]
Indeed there are some schedules marked as "CANCELED" in the trip_update... However, there are so few that I think it might not be exhaustive even for all the companies that supported GTFS-RT... I will take a look into http://fahrplan.sbb.ch/bin/help.exe/en?tpl=showmap_external

Thanks for the info

**Comment by ShiroYacha** [comment]
Hi,

I am also looking at platform changes. If I understand correctly, platform changes are not yet integrated in transport.opendata.ch? I am looking into GTFS-RT, but it seems that this is not explicitely reported either there (actually, if they list a stop_id, and there is a platform change, the data seems to be correct judging by the platform number in the id, e.g. 8502202:0:5; however, it seems that there is no trigger for this type of changes...)...

How do you deal with these kinds of changes? 

Thanks!

**Comment by helbling** [comment]
hi
I have enabled track changes as experimental opt-in on the timetable.search.ch API.
If you add &show_trackchanges=1 changed tracks are considered. Changed tracks are returned in the same field normal planned tracks are returned and an exclamation mark is appended.

**Comment by hb9tst** [comment]
Are there news about this? Two years passed but it still looks like the real time data is not available (neither on transport.search.ch nor via the opendata API). For instance, tonight the IC5 from Neuchâtel to Lausanne is cancelled (which appears on the SBB timetable), but this is not visible from transport.search.ch...
![Cancelled train on SBB web site](https://user-images.githubusercontent.com/30394208/74966236-fe426800-5416-11ea-894c-5b1af5fe98b8.png)
![Same train on transport.search.ch, no trace of cancellation](https://user-images.githubusercontent.com/30394208/74966244-026e8580-5417-11ea-9b8c-e2ae9eff2d95.png)

**Comment by schafran** [comment]
Are there any news on this? timetable.search.ch seems to have the information about cancelled connections, however I see no info about cancellations on the `connections` api endpoint.
Is there any other way I can find whether a connection is cancelled or not? Delay seems to be working fine.

**Comment by beeb** [comment]
Any news on this @fabian ? It's been years and there doesn't seem to be a way to check if a connection is cancelled or not. Also no-one replies to issues and comments so it's looking very dead around here.

**Comment by fabian** [comment]
Sorry, no news I'm aware of.

**Comment by helbling** [comment]
If you use the `search.ch` timetable API, you can see cancelled trips. They are marked with `X` in the `dep_delay` field.

Example:
https://search.ch/fahrplan/api/route.json?from=Zell+(Wiesenthal)&to=Basel+Bad+Bf&show_delays=1

```
{
  "count": 6,
  "min_duration": 2460,
  "rawtime": 2460,
  "max_duration": 4680.0,
  "maxtime": 4680.0,
  "connections": [
    {
      "duration": 2580,
      "from": "Zell im Wiesental",
      "departure": "2025-03-26 09:30:00",
      "dep_delay": "X",
      "to": "Basel Bad Bf",
      "arrival": "2025-03-26 10:13:00",
      "arr_delay": "+0",
      "is_main": true,
      "infeasibility_level": 3,
      "legs": [
[...]
```
